### PR TITLE
Add CE Secure Marker

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
New column for identifying if CE case is a Secure Establishment

# What has changed
New CE_SECURE field name
Included secureEstablishment
Tests

# How to test?
Run tests locally
Then run ATs with our repos on card

# Links
https://collaborate2.ons.gov.uk/confluence/display/CRM/Address+Frame+specification+2021?focusedCommentId=36183836#comment-36183836
https://trello.com/c/BAI9qOOC/619-rmc-341-add-secure-marker-case-api-and-field-8

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[RMC-341 Add secure marker - case api and field (8)](https://trello.com/c/BAI9qOOC/619-rmc-341-add-secure-marker-case-api-and-field-8)

File: generate_sample_file.py [hunk 1]
```diff
@@ -205,7 +210,8 @@ def _write_row(self, writer, sequential_arid, treatment_code, address_type, expe
             'TREATMENT_CODE': treatment_code["treatment_code"],
             'FIELDCOORDINATOR_ID': '',
             'FIELDOFFICER_ID': '',
-            'CE_EXPECTED_CAPACITY': expected_capacity
+            'CE_EXPECTED_CAPACITY': expected_capacity,
+            'CE_SECURE': self.get_random_ce_secure()
```
Comment (LukeLoze1) [inline generate_sample_file.py]: I'd prefer this to be tied to when we Create CE and SPG only to create more realistic data.

File: validate_sample.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def __init__(self):
             'FIELDOFFICER_ID': [max_length(10)],
             'TREATMENT_CODE': [mandatory(), in_set(self.TREATMENT_CODES)],
             'CE_EXPECTED_CAPACITY': [numeric(), max_length(4)],
+            'CE_SECURE': [mandatory(), max_length(2)]
```
Comment (LukeLoze1) [inline validate_sample.py]: Should this be 1 or 0 be max_length(2)?

**Comment by LukeLoze1** [comment]
Couple of minor comments/questions

**Review by LukeLoze1 — CHANGES_REQUESTED** [review]
few questions

File: validate_sample.py (hunk 2 above)
Comment (AdamHawtin) [reply validate_sample.py]: Should be `in_set({'0','1'})` since those are the only two valid values

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Agree with Lukes comment regarding generating the CE SECURE values, also should be specifically checking for 0 or 1 in the column validators

File: generate_sample_file.py (hunk 1 above)
Comment (NickGrantX) [reply generate_sample_file.py]: That's a massive "nice to have". This random sample doesn't even create any random CE cases yet. We should just change it to zero, and not have it random if you're going to fuss.

Comment (LeoKHoward) [reply generate_sample_file.py]: The sample does create random CE cases (line 147) but an if statement has now been included to randomly assign a 0/1 value in CE_SECURE is the address_type is 'CE'

**Review by LukeLoze1 — APPROVED** [review]

**Review by AdamHawtin — APPROVED** [review]

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Pull request merged**